bitcoin/src/blockdata/transaction.rs

```diff
@@ -232,6 +232,16 @@ impl TxIn {
     const BASE_WEIGHT: Weight =
         Weight::from_vb_unwrap(OutPoint::SIZE as u64 + Sequence::SIZE as u64);
 
+    /// Creates a new empty [`TxIn`] with the provided [`Sequence`].
+    pub fn new(sequence: Sequence) -> TxIn {
+        TxIn {
+            previous_output: OutPoint::null(),
```

Comment:
This creates coinbase input and the vast majority of transactions are not coinbase. This should be called `coinbase`.

I don't think `TxIn::new` has any value whatsoever. There's no reasonable default for `TxIn` anyway and the correct `new` function would have to accept every field as a parameter which is much worse than initializing a struct.